osu.Game/Beatmaps/BeatmapManager_WorkingBeatmap.cs

```diff
@@ -33,6 +33,9 @@ public BeatmapManagerWorkingBeatmap(IResourceStore<byte[]> store, TextureStore t
 
             protected override IBeatmap GetBeatmap()
             {
+                if (BeatmapInfo.Path == null)
+                    return BeatmapInfo?.Ruleset.CreateInstance().CreateBeatmapConverter(new Beatmap { BeatmapInfo = BeatmapInfo }).Beatmap;
```

Comment:
Why not `null`? This looks to me like some code is expecting this func to never return `null`, but that isn't always the case.

Reply:
This is what makes a new beatmap for consumption. Should it be in a different place?

Reply:
Where does that happen? Doesn't BeatmapManager create a file anyway?

Reply:
I don't follow. Where are you seeing it "create a file"?

Reply:
Ah, I was wrong, it only does that on `Save()`.

In that case, return the `new Beatmap() {...}` part?

Reply:
without the conversion? for some reason i thought that was required..

Reply:
Yeah. Conversion is only done by `WorkingBeatmap.GetPlayableBeatmap()`

Reply:
have you tested this works? because as i say there was 100% a reason i did the conversion.

Reply:
I've tried removing the conversion here and nothing seemed to catch fire - unsure whether I've missed some scenario, though. Mostly just entered and exited the editor a couple of times, saved, exported a few times. The two added tests also pass with the suggested change.

Reply:
It does look to work, weird. I must have fixed whatever was requiring this in the first place 👍 .